lib/phase/functional.js

```diff
@@ -315,6 +315,13 @@ function validateStages(stages, jobs) {
         });
     });
 
+    // If jobs in stages have sourcePaths set, throw error
+    stageJobNames.forEach(jobName => {
```

Comment:
We should also perform the same check on stage teardown job

Reply:
Created PR to modify to validate with data-schema.
https://github.com/screwdriver-cd/data-schema/pull/592